pandas/tseries/period.py

```diff
@@ -271,10 +271,13 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
+
+        if not com.is_integer_dtype(values):
```

Comment:
Yes internal data must be `int` dtype. So it should be either coerced / raised.

I think coercing float from int is OK, as long as it is consistent. I haven't noticed float coercion in below case. Let us split it to separate issue.

```
pd.PeriodIndex([1.1], freq='M')
# ValueError: Given date string not likely a datetime.
pd.PeriodIndex(np.array([1.1]), freq='M')
# PeriodIndex(['1970-02'], dtype='int64', freq='M')
```

ATM, slight modification of @jereback suggestion should work?

```
if com.is_integer_dtype(values) or len(values) == 0:
     values = np.array(values, dtype='int64', copy=False)
else:
   return PeriodIndex...
```